2017/1/day1.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+func intFromRune(r rune) (int64, error) {
+	s_r := string(r)
+	v, err := strconv.ParseInt(s_r, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	return v, nil
+}
+
+func nextPart2(i int, input string) int {
+	// Last char is a line feed
+	real_len := len(input) - 1
+	// The length of the input will always be divisible by 2
+	half := real_len / 2
+	return (i + half) % real_len
+}
+
+func nextPart1(i int, input string) int {
+	// Last char is a line feed
+	if i == len(input)-2 {
+		return 0
+	} else {
```

Comment:
The go linter gets upset when the `if` block ends with a `return` statement and there is an `else`/`else if` clause.

```go
func nextPart1(i int, input string) int {
        if i == len(input)-2 {
                return 0
        }

        return i + 1
}
```

Not sure if vscode shows you linter errors by default?

Reply:
I may have been using vim :sweat_smile: and obviously do not have my plugins setup properly :thinking: